contracts/src/gateway/router/CheckpointingFacet.sol

```diff
@@ -109,13 +142,19 @@ contract CheckpointingFacet is GatewayActorModifiers {
         if (!exists) {
             revert CheckpointNotCreated();
         }
-        LibQuorum.addQuorumSignature({
+        bool isQuorumReached = LibQuorum.addQuorumSignature({
             self: s.checkpointQuorumMap,
             height: height,
             membershipProof: membershipProof,
             weight: weight,
             signature: signature
         });
+
+        // if a new quorum is reached, the checkpoint should be forwarded to
+        // the parent and a new checkpoint with a bigger height should be created
+        if (isQuorumReached) {
+            s.prevBottomUpCheckpointHeight = height;
```

Comment:
I don't think this is correct here. Checkpoints are created any time the conditions are right: 
* Too many messages in the buffer
* Some messages in the buffer have been waiting too long
* Too much time has passed without a chekpoint

Under any of these conditions the ledger creates a checkpoint. The next checkpoint should point at that one, the `lastbottomUpCheckpointHeight` should be updated as soon as the checkpoint is created. 

Then, validators start submitting signatures to the checkpoints, but there can be any number of them that need signing, we don't know which one reaches quorum first. They could go in opposite order of creation. This variable should not be updated just because some checkpoint reached a quorum. An event should be emitted, maybe the checkpoint flagged as completed, but that's it.

Reply:
Yeah, understand. Initlally I was wondering if there are chances of validator not agreeing on the quorum, so that only a quorum is reached, we can conclude that the checkpoint is created and set previous checkpoint height.